pkg/printers/tablegenerator.go

```diff
@@ -115,11 +115,6 @@ func (h *HumanReadableGenerator) GenerateTable(obj runtime.Object, options Print
 			table.SelfLink = m.GetSelfLink()
 		}
 	}
-	// TODO(seans3): Remove the following decorateTable call. This should only be
-	// called in the table printer.
-	if err := decorateTable(table, options); err != nil {
```

Comment:
since we're removing this, GenerateTable will no longer honor most of the fields in PrintOptions. That's not a problem today, since the only caller only uses the `Wide` and `NoHeaders` options, but it makes the API misleading.

Suggest making a GenerateOptions struct containing only the `Wide` and `NoHeaders` options and changing the signature to `GenerateTable(obj runtime.Object, options GenerateOptions)`:

```diff
diff --git a/pkg/printers/tablegenerator.go b/pkg/printers/tablegenerator.go
index b45c585bfb4..04d7918262d 100644
--- a/pkg/printers/tablegenerator.go
+++ b/pkg/printers/tablegenerator.go
@@ -27,9 +27,14 @@ import (
        utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 )

+type GenerateOptions struct {
+       NoHeaders bool
+       Wide      bool
+}
+
 // TableGenerator - an interface for generating metav1beta1.Table provided a runtime.Object
 type TableGenerator interface {
-       GenerateTable(obj runtime.Object, options PrintOptions) (*metav1beta1.Table, error)
+       GenerateTable(obj runtime.Object, options GenerateOptions) (*metav1beta1.Table, error)
 }

 // PrintHandler - interface to handle printing provided an array of metav1beta1.TableColumnDefinition
@@ -71,14 +76,15 @@ func (h *HumanReadableGenerator) With(fns ...func(PrintHandler)) *HumanReadableG
 // GenerateTable returns a table for the provided object, using the printer registered for that type. It returns
 // a table that includes all of the information requested by options, but will not remove rows or columns. The
 // caller is responsible for applying rules related to filtering rows or columns.
-func (h *HumanReadableGenerator) GenerateTable(obj runtime.Object, options PrintOptions) (*metav1beta1.Table, error) {
+func (h *HumanReadableGenerator) GenerateTable(obj runtime.Object, options GenerateOptions) (*metav1beta1.Table, error) {
        t := reflect.TypeOf(obj)
        handler, ok := h.handlerMap[t]
        if !ok {
                return nil, fmt.Errorf("no table handler registered for this type %v", t)
        }

-       args := []reflect.Value{reflect.ValueOf(obj), reflect.ValueOf(options)}
+       printOptions := PrintOptions{NoHeaders: options.NoHeaders, Wide: options.Wide}
+       args := []reflect.Value{reflect.ValueOf(obj), reflect.ValueOf(printOptions)}
        results := handler.printFunc.Call(args)
        if !results[1].IsNil() {
                return nil, results[1].Interface().(error)
```

I think only a single caller will need to be updated (other than the unit tests in `pkg/printers/internalversion/printers_test.go`)